test/like.test.js

```diff
@@ -0,0 +1,312 @@
+        if (error) done(error);
+        expect(res).have.status(200);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('message', 'status', 'data');
+        expect(res.body.message).to.deep.equal('Successfully disliked');
+        expect(res.body.data).to.have.keys('ArticleSlug', 'status', 'claps');
+        expect(res.body.data.claps).to.deep.equal(0);
+        expect(res.body.data.status).to.deep.equal('dislike');
+        done();
+      });
+  });
+  it('should return all Dislike to a certain article ', (done) => {
+    chai
+      .request(server)
+      .get('/api/v1/likes/dislikes/fakeslug')
+      .set('Authorization', usertoken)
+      .end((error, res) => {
+        if (error) done(error);
+        expect(res).have.status(200);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('message', 'data', 'status');
+        expect(res.body.message).to.deep.equal('Successfully dislike retrieved');
+        expect(res.body.data).to.have.keys('dislikes');
+        done();
+      });
+  });
+  it('should Unlike Article due to dislike secomd time ', (done) => {
+    chai
+      .request(server)
+      .put('/api/v1/likes/dislike/fakeslug')
+      .set('Authorization', usertoken)
+      .end((error, res) => {
+        if (error) done(error);
+        expect(res).have.status(200);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('message', 'status', 'data');
+        expect(res.body.message).to.deep.equal('Successfully disliked');
+        expect(res.body.data).to.have.keys('ArticleSlug', 'status', 'claps');
+        expect(res.body.data.claps).to.deep.equal(0);
+        expect(res.body.data.status).to.deep.equal('neutral');
+        done();
+      });
+  });
+
+
+  it('should not unlike unexisting article', (done) => {
+    chai
+      .request(server)
+      .put('/api/v1/likes/unlike/fakesl')
+      .set('Authorization', usertoken)
+      .end((error, res) => {
+        if (error) done(error);
+        expect(res).have.status(404);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('message', 'status');
+        expect(res.body.message).to.deep.equal('post not found');
+        done();
+      });
+  });
+  it('Should successfully clap ', (done) => {
+    chai
+      .request(server)
+      .put('/api/v1/likes/clap/fakeslug')
+      .set('x-access-token', mikeToken)
+      .end((error, res) => {
+        if (error) done(error);
+        expect(res).have.status(200);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('message', 'status', 'data');
+        expect(res.body.message).to.deep.equal('Successfully claped');
+        expect(res.body.data).to.have.keys('ArticleSlug', 'status', 'claps');
+        expect(res.body.data.claps).to.deep.equal(1);
+        expect(res.body.data.status).to.deep.equal('like');
+        done();
+      });
+  });
+  it('Should successfully clap ', (done) => {
```

Comment:
This test is a duplicate of the one above. Consider removing it.

Reply:
> This test is a duplicate of the one above. Consider removing it.

thanks, noa for your feedback however I have to test if someone can clap so many times that  means I have to clap multiple time